src/Components/DesignAdjustments.tsx

```diff
@@ -39,17 +41,49 @@ export const DesignAdjustments = connect(
 
     const fontHeadlineWeight = root.get('siteFontHeadlineWeight') || '500'
     styles.push(`--jr-headline-font-weight: ${fontHeadlineWeight};`)
+    const fontHeadline = root.get('siteFontHeadline')[0]
+    if (isFont(fontHeadline)) {
+      styles.push(`--jr-headline-font-family: ${fontHeadline.get('family')};`)
+    }
+
     const fontBodyWeight = root.get('siteFontBodyWeight') || '500'
     styles.push(`--bs-body-font-weight: ${fontBodyWeight};`)
+    const fontBody = root.get('siteFontBody')[0]
+    if (isFont(fontBody)) {
+      styles.push(`--bs-body-font-family: ${fontBody.get('family')};`)
+      styles.push(`--toastify-font-family: ${fontBody.get('family')};`)
+    }
+
     return (
       <>
         <Helmet>
           {/* @ts-expect-error helmet bug: https://github.com/nfl/react-helmet/issues/344*/}
           <body style={styles.join(' ')}></body>
         </Helmet>
+
+        <CustomFonts root={root} />
         {children}
       </>
     )
   },
   { loading: () => null },
 )
+
+const CustomFonts = connect(function CustomFonts({
+  root,
+}: {
+  root: HomepageInstance
+}) {
+  return (
+    <>
+      {[...root.get('siteFontHeadline'), ...root.get('siteFontBody')]
+        .filter(
+          (font, index, self) =>
+            index === self.findIndex((f) => f.id() === font.id()),
+        )
```

Comment:
* [ ] From reading the filter code I have no idea what it does. Maybe extract to a named function or put a comment?